### ISSUE TITLE
Codec Support Query: Basis Universal

### ISSUE THREAD

**Issue by SignpostMarv** [GoogleChromeLabs/squoosh#642]
**Is your feature request related to a problem? Please describe.**
Just noticed ["Basis Universal"](https://github.com/binomialLLC/basis_universal) pop up on [my twitter feed](https://mobile.twitter.com/Lady_Ada_King/status/1144546463365681152)
> ...a very compressed image format that are smaller to download will allow 6-8 times more textures in our WebGL scenes, enabling us to build richer worlds at higher fidelity

Was wondering if (in general) image formats that are targeted mostly at WebGL are in or out-of-scope for Squoosh, and (specifically) if this would be something that Squoosh could support?

**Describe the solution you'd like**
Yes/No on "is this format out-of-scope for Squoosh" ?

**Does other service/app have this feature?**
* Three.js

**Additional context**
* [Using Basis Textures in Three.js](https://medium.com/samsung-internet-dev/using-basis-textures-in-three-js-6eb7e104447d) by @adarosecannon
* #608

**Comment by surma** [comment]
I’ll take a look at this :)

**Comment by mnordine** [comment]
Any update on this?

**Comment by surma** [comment]
The licenses/IP/patents on these formats are unclear to me, so I can’t really ship the to Squoosh without having a lawyer look at them :-/

**Comment by mnordine** [comment]
License is here: https://github.com/BinomialLLC/basis_universal/blob/master/LICENSE

Seems ok?

**Comment by richgel999** [comment]
Basis Universal uses the Apache 2.0 license. It's a 100% free open source library that may be used for any purpose, including commercial purposes. The authors and Binomial LLC have no patents whatsoever, and we are 100% against software patents. Binomial is officially a member of the Khronos Group. All the formats supported by Basis Universal are documented in the official Khronos Texture Format Specification.

**Comment by richgel999** [comment]
The Khronos Texture Format Specification is here:
https://www.khronos.org/opengles/sdk/tools/KTX/file_format_spec/

The transcoder was written using information in this open specification. From a licensing perspective, as long as you aren't creating GPU hardware you frankly don't have anything to worry about whatsoever.

**Comment by richgel999** [comment]
For ASTC in particular, our use case falls under the ASTC license:
https://github.com/ARM-software/astc-encoder/blob/master/LICENSE.md

**Comment by romainguy** [comment]
@surma I can give you more context (my ldap is romainguy@)

**Comment by surma** [comment]
Had a meeting @romainguy, we’ll look into this codec :)

**Comment by mnordine** [comment]
Any update on this?

**Comment by surma** [comment]
Yeah, I briefly looked into this, but didn’t get too far. We had higher-priority features. I still want to do this tho. Happy for someone else to tackle this, btw. Squoosh v2 is significantly easier to contribute to.
